docs/src/sequencepaxos/storage.md

```diff
@@ -0,0 +1,208 @@
+# Storage
+You are free to use any storage implementation with `SequencePaxos`. The only requirement is that it implements the `Storage` trait. OmniPaxos includes the package `omnipaxos_storage` which provides two types of storage implementation, depending on the users need for fast reads and writes or persistance: `MemoryStorage` and `PersistentStorage`
+
+## MemoryStorage
+An in-memory storage implementation, `MemoryStorage` offers fast reads and writes. This storage type will be used in our examples. For simplicity, we leave out some parts of the implementation for now (such as [Snapshots](../compaction.md)).
+```rust,edition2018,no_run,noplaypen
+    // from the module omnipaxos_storage::memory_storage
+    #[derive(Clone)]
+    pub struct MemoryStorage<T, S>
+    where
+        T: Entry,
+        S: Snapshot<T>,
+    {
+        /// Vector which contains all the replicated entries in-memory.
+        log: Vec<T>,
+        /// Last promised round.
+        n_prom: Ballot,
+        /// Last accepted round.
+        acc_round: Ballot,
+        /// Length of the decided log.
+        ld: u64,
+        ...
+    }
+
+    impl<T, S> Storage<T, S> for MemoryStorage<T, S>
+    where
+        T: Entry,
+        S: Snapshot<T>,
+    {
+        fn append_entry(&mut self, entry: T) -> u64 {
+            self.log.push(entry);
+            self.get_log_len()
+        }
+
+        fn append_entries(&mut self, entries: Vec<T>) -> u64 {
+            let mut e = entries;
+            self.log.append(&mut e);
+            self.get_log_len()
+        }
+
+        fn append_on_prefix(&mut self, from_idx: u64, entries: Vec<T>) -> u64 {
+            self.log.truncate(from_idx as usize);
+            self.append_entries(entries)
+        }
+
+        fn set_promise(&mut self, n_prom: Ballot) {
+            self.n_prom = n_prom;
+        }
+
+        fn set_decided_idx(&mut self, ld: u64) {
+            self.ld = ld;
+        }
+
+        fn get_decided_idx(&self) -> u64 {
+            self.ld
+        }
+
+        fn set_accepted_round(&mut self, na: Ballot) {
+            self.acc_round = na;
+        }
+
+        fn get_accepted_round(&self) -> Ballot {
+            self.acc_round
+        }
+
+        fn get_entries(&self, from: u64, to: u64) -> &[T] {
+            self.log.get(from as usize..to as usize).unwrap_or(&[])
+        }
+
+        fn get_log_len(&self) -> u64 {
+            self.log.len() as u64
+        }
+
+        fn get_suffix(&self, from: u64) -> &[T] {
+            match self.log.get(from as usize..) {
+                Some(s) => s,
+                None => &[],
+            }
+        }
+
+        fn get_promise(&self) -> Ballot {
+            self.n_prom
+        }
```

Comment:
Please update this according to the changes you have made. For example `get_entries` now returns a vector instead of slice.